src/otp_sdk/get_refresh_token.js

```diff
@@ -0,0 +1,47 @@
+const verifyToken = require('../helpers/verify_token.js');
+const axios = require('axios');
+const qs = require('querystring');
+const jwt = require('jsonwebtoken');
+const { oauthTokenBaseURL } = require('../constants/urls.json');
+
+module.exports = (req, res) => {
+  const requestToken = token => {
+    console.log(1);
+    const tokenQueries = {
+      grant_type: 'refresh_token',
+      refresh_token: token.refresh_token,
+      client_id: process.env.CLIENT_ID,
+      client_secret: process.env.CLIENT_SECRET,
+      redirect_uri: process.env.REDIRECT_URI,
+    };
+
+    const options = {
+      method: 'post',
+      baseURL: oauthTokenBaseURL,
+      data: qs.stringify(tokenQueries),
+    };
+
+    return new Promise(async (resolve, reject) => {
+      try {
+        const apiTokenResponse = await axios(options);
+        const { access_token, refresh_token } = apiTokenResponse.data;
+        const token = jwt.sign(
+          { access_token, refresh_token },
+          process.env.JWT_SECRET,
+        );
+        res.clearCookie('access');
+        res.cookie('access', token, { maxAge: 604800000 });
+        resolve(access_token);
+      } catch (e) {
+        reject(e);
+      }
+    });
+  };
+
+  return new Promise(async (resolve, reject) => {
```

Comment:
Okay,
This promise wrapper is unnecessary

also,
 
This is the final change I will request,

There should be a clear distinction between something that is a controller, and something that is not a controller.

This file should be more like a plain function, it should take a bunch or arguments, and return a promise.

Ie, remove req and res from this function, and deal with these outcomes inside a controller. This may require a bit of work